src/clusterfuzz/_internal/bot/tasks/utasks/analyze_task.py

```diff
@@ -406,29 +420,58 @@ def handle_build_setup_error(output):
         wait_time=testcase_fail_wait)
     return
   data_handler.mark_invalid_uploaded_testcase(
-      output.testcase, output.testcase_upload_metadata, 'Build setup failed')
+      testcase, testcase_upload_metadata, 'Build setup failed')
 
 
 # pylint: disable=no-member
 HANDLED_ERRORS = [
     uworker_msg_pb2.ErrorType.ANALYZE_NO_CRASH,
     uworker_msg_pb2.ErrorType.ANALYZE_BUILD_SETUP,
     uworker_msg_pb2.ErrorType.ANALYZE_NO_REVISIONS_LIST,
-    uworker_msg_pb2.ANALYZE_NO_REVISION_INDEX,
+    uworker_msg_pb2.ErrorType.ANALYZE_NO_REVISION_INDEX,
     uworker_msg_pb2.ErrorType.UNHANDLED,
 ] + setup.HANDLED_ERRORS
 
 # pylint: enable=no-member
 
 
+def _update_testcase(output):
+  """Updates the testcase using the info passed from utask_main."""
+  testcase = data_handler.get_testcase_by_id(output.uworker_input.testcase_id)
+  analyze_task_output = output.analyze_task_output
+  testcase.crash_revision = analyze_task_output.crash_revision
+
+  if analyze_task_output.absolute_path:
+    testcase.absolute_path = analyze_task_output.absolute_path
+
+  if analyze_task_output.minimized_arguments:
+    testcase.minimized_arguments = analyze_task_output.minimized_arguments
+
+  if analyze_task_output.crash_stacktrace:
+    testcase.crash_stacktrace = analyze_task_output.crash_stacktrace
+
+  if analyze_task_output.crash_info_set:
+    testcase.http_flag = analyze_task_output.http_flag
+    testcase.crash_type = analyze_task_output.crash_type
+    testcase.crash_address = analyze_task_output.crash_address
+    testcase.crash_state = analyze_task_output.crash_state
+    testcase.security_flag = analyze_task_output.security_flag
+    if testcase.security_flag:
+      testcase.security_severity = analyze_task_output.security_severity
+
+  testcase.one_time_crasher_flag = analyze_task_output.one_time_crasher_flag
+  testcase.put()
+
+
 def utask_postprocess(output):
   """Trusted: Cleans up after a uworker execute_task, writing anything needed to
   the db."""
+  _update_testcase(output)
   if output.error_type is not None:
     uworker_handle_errors.handle(output, HANDLED_ERRORS)
     return
-  testcase = output.testcase
-  testcase_upload_metadata = output.testcase_upload_metadata
+  testcase = data_handler.get_testcase_by_id(output.uworker_input.testcase_id)
+  testcase_upload_metadata = output.uworker_input.testcase_upload_metadata
```

Comment:
Can we remove the `testcase_upload_metadata` field from the Output proto?

Reply:
`testcase_upload_metadata` is no longer part of the output here. It is accessed through the `uworker_input` which is added to the output after `utask_main`

Reply:
Right. Can we remove the proto field completely? I.e. this: https://github.com/google/clusterfuzz/blob/cefc990128d95fa570bf84bed2494fcbc48aba23/src/clusterfuzz/_internal/protos/uworker_msg.proto#L181

Reply:
I think so.

Reply:
Removed, I'll also remove the testcase and variant UworkerEntityWrapper once the other PR lands.